driver-async/src/main/com/mongodb/async/client/MongoDatabase.java

```diff
@@ -138,9 +136,47 @@
     /**
      * Gets the names of all the collections in this database.
      *
-     * @param callback the callback that is passed the names of all the collections in this database
+     * @return an iterable containing all the names of all the collections in this database
+     */
+    MongoIterable<String> listCollectionNames();
+
+    /**
+     * Finds all the collections in this database.
+     *
+     * @return an iterable containing all the collections in this database
+     * @mongodb.driver.manual reference/command/listCollections listCollections
+     */
+    MongoIterable<Document> listCollections();
+
+    /**
+     * Finds all the collections in this database.
+     *
+     * @param clazz the class to decode each document into
+     * @param <C>   the target document type of the iterable.
+     * @return an iterable containing all the collections in this database
+     * @mongodb.driver.manual reference/command/listCollections listCollections
+     */
+    <C> MongoIterable<C> listCollections(Class<C> clazz);
+
+    /**
+     * Finds the collections in this database.
+     *
+     * @param filter the query filter
+     * @return an iterable containing all the collections in this database
+     * @mongodb.driver.manual reference/command/listCollections listCollections
+     */
+    MongoIterable<Document> listCollections(Object filter);
```

Comment:
Yes but at the moment we only have batchSize as an extra option and it is only applied in 2.8.  I thought this was more akin to the fluentFind which we I could implement for listCollections and listIndexes and then we can expand in the future.